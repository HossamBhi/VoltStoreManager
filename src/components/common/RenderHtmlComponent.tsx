import React from 'react';
import {useWindowDimensions} from 'react-native';
import RenderHtml from 'react-native-render-html';

export default ({html, textColor}: {html?: string; textColor?: string}) => {
  const {width} = useWindowDimensions();

  return (
    <RenderHtml
      contentWidth={width}
      source={{
        html: `<body dir="rtl" ><p style="padding: 0px; margin: 0;" dir="rtl" >${html}</p></body>`,
      }}
      tagsStyles={{
        body: {
          paddingTop: 0,
          marginTop: 0,
          color: textColor || '#fff',
        },
        p: {
          paddingBottom: 0,
          marginBottom: 0,
          paddingTop: 0,
          marginTop: 5,
        },
      }}
    />
  );
};
