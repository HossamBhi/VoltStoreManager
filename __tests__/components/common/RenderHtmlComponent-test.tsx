import React from 'react';
import {create} from 'react-test-renderer';
import RenderHtmlComponent from '../../../src/components/common/RenderHtmlComponent';

it('Render Html Component', () => {
  const CHtml = create(<RenderHtmlComponent />);
  expect(CHtml.toJSON()).toMatchSnapshot();
});
