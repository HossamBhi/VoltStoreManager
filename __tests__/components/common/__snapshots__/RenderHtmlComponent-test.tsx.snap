// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Html Component 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
      undefined,
    ]
  }
  testID="html"
>
  <View
    style={
      Array [
        Object {
          "marginTop": 0,
          "paddingTop": 0,
        },
        undefined,
        undefined,
      ]
    }
    testID="body"
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          undefined,
          undefined,
        ]
      }
      testID="p"
    >
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        undefined
      </Text>
    </View>
  </View>
</View>
`;
